database/redis/notification.go

```diff
@@ -98,7 +130,119 @@ func (connector *DbConnector) removeNotifications(notifications []*moira.Schedul
 }
 
 // FetchNotifications fetch notifications by given timestamp and delete it
-func (connector *DbConnector) FetchNotifications(to int64) ([]*moira.ScheduledNotification, error) {
+func (connector *DbConnector) FetchNotifications(to int64, limit int64) ([]*moira.ScheduledNotification, error) {
+	// No limit
+	if limit == 0 {
+		return connector.fetchNotificationsNoLimit(to)
+	}
+
+	count, err := connector.notificationsCount(to)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hope count will be not greater then limit when we call fetchNotificationsNoLimit
+	if count == nil {
```

Comment:
Do we really have situations when `count` can be nil?

Reply:
Even is `count` can be nil think we need to move this check to `connector.notificationsCount`

Reply:
Done